appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/OpenSourceVulnerabilityClient.java

```diff
@@ -45,7 +45,19 @@ class OpenSourceVulnerabilityClient {
         rateLimiter = new RateLimiter(ratePerSecond);
     }
 
-    QueryBatchResponse queryBatch(QueryBatchRequestPayload payload) {
+    /**
+     * Do a batch query to the OSV API to check vulnerabilities of multiple
+     * software packages in a single request.
+     *
+     * @param payload
+     *            the payload to send to OSV API server with all packages to
+     *            check.
+     * @return a response with a list of vulnerabilities.
+     * @throws AppSecException
+     *             if there is a connection issue with the OSV API server.
+     */
+    QueryBatchResponse queryBatch(QueryBatchRequestPayload payload)
+            throws AppSecException {
```

Comment:
Should we add the `throws AppSecException` to the `queryVulnerability()` method too? It also uses the `createHttpURLConnection()`.

Reply:
Sure.